### DISCUSSION TITLE
RustDesk is facing severe performance issues during transmission.

### DISCUSSION THREAD

**Discussion by Arc-2023** [rustdesk/rustdesk#6888] [General]
### Bug Description

**Issues:**
1. Even in successful punching through (confirmed by direct IP connection with latency below 10ms), there is still severe stuttering in the transmission of video images. Additionally, the custom configuration settings for bitrate and framerate seem ineffective, with a bitrate limit of approximately 600KB/s.

2. The problem described in point 1 is more pronounced when using a mobile device as the controlled endpoint, where even simple actions like scrolling result in a framerate drop below 5FPS. The symptoms are less severe when controlling a computer, but rapid window shaking also leads to significant frame drops.

### How to Reproduce

**Testing Environment:**
0. Ryzen 5600H + RTX3050Ti + Windows 11 22h2
1. Confirmed that network bandwidth is not the bottleneck (Moonlight, using Tailscale as well, achieves much higher bandwidth).
2. Confirmed that hardware is not the bottleneck (CPU and GPU usage are not maxed out; Moonlight can utilize more resources).
**Deployment Method:**
hbbr and hbbs are deployed using K3S, with the specification `hostNetwork: true` in the deployment to directly listen on the node network, thus avoiding SNAT.

*Both the Android and Windows versions are built using the nightly build.*

**Testing Methods:**
1. Connect using an ID, allowing RustDesk to punch through and act as an intermediary.
2. Use Tailscale to create a network across different locations, utilizing DERP for punching through and acting as an intermediary.

### Expected Behavior

For local network transmission, there should be no frame drops at 30FPS, even at lower bitrates (blurriness instead of dropped frames).

### Operating system(s) on local side and remote side

win11 22h2-> android 13; android 13-> win11 22h2;win10-> android 13

### RustDesk Version(s) on local side and remote side

1.2.4(nightly build)-> docker latest image

### Screenshots

![image](https://github.com/rustdesk/rustdesk/assets/64178177/d63b5633-d6e7-4d5e-987f-f277f1d9c2a5)
![image](https://github.com/rustdesk/rustdesk/assets/64178177/0bf15cab-b8b3-4b80-b4c9-cb4964eefc63)

### Additional Context
The noteworthy point is that when controlling the phone with the mouse, the actions can be instantly seen on the phone; however, there is a significant delay and frame loss when the screen changes are transmitted back to the computer.

**Comment by 21pages** [comment]
bitrate limit is reproduced only on av1 and  frame drops is allowed but rapid window shaking causing significant frame drops is device specific.

https://github.com/rustdesk/rustdesk/assets/14891774/7c79135d-2478-4119-9921-b6daccdcb5ad

https://github.com/rustdesk/rustdesk/assets/14891774/3900c34f-db6e-4e6f-912c-bed3cd53ec1a

**Comment by Arc-2023** [comment]
> bitrate limit is reproduced only on av1 and frame drops is allowed but rapid window shaking causing significant frame drops is device specific.
> 
>  rustdesk_aP6jZQYUtH.mp4 
>  4b23feb24b3acafe100d27576b250256.mp4

Sorry for testing encoding ways incompletely. When switching to VP8/VP9, the smoothness and response time improve significantly. However, in the context of mobile phones serving as controlled devices, performance remains subpar, and the overall user experience remains unsatisfactory.

While AV1 may be particularly unfriendly for machines lacking dedicated hardware circuits for AV1 encoding, the issue of poor performance is not exclusive to AV1.

![Image](https://github.com/rustdesk/rustdesk/assets/64178177/d0286b1b-a1e0-4618-8d1c-ef9ce377c59d)

In the AV1 and VP9 modes, although the experience is quite poor on Samsung Tab S8 (8Gen1), one could tentatively attribute it to encoding bottlenecks. However, in the VP8 mode, the visible frame rate remains extremely low, leading to significant stuttering and frame drops during desktop scrolling.

It is worth noting that within the first 1-2 seconds of establishing a connection, the visuals appear relatively smooth. However, after this initial period, frame drops become apparent, with a frame rate not exceeding 15 frames per second (FPS) and a bitrate not surpassing 600 KB/s.

But it doesn't happen on other devices (Redmi K50 Dimensity 8100).


I noticed some people suggesting that due to RustDesk's positioning as office software, it actively reduces bitrate during streaming. However, I still believe that providing the lowest possible latency and high-quality visuals when the user desires is crucial, even if it means consuming significant network bandwidth.

**Comment by mahergreg** [comment]
I commented in a similar [issue here](https://github.com/rustdesk/rustdesk/discussions/10050#discussioncomment-12893816)
My problem was Windows scaling caused big lag. You need to allow the app to handle its own scaling and try to match the resolutions of the host & client monitors.

Note, there "seems" to be a halving of the FPS when using custom mode (possibly a different issue); i.e. set it to 60 FPS and you get 30FPS, set it to 120 FPS and you get 60 FPS.

Hope this helps.

**Comment by Frankstar** [comment]
im facing a similar error.
everything was good till i connect an ELO Touch where the native resolution ist 1024x768.
the image is super fast! but the input lags horrible. (3 - 5 seconds delay)

Draging a box from top to bottom and than back bottom to top:

https://github.com/user-attachments/assets/91b76dc2-8475-4877-af42-0b9006b57002


played aroung with fps and bitrate - but that didnt change anything in a good way.
